src/storage/src/hummock/compactor/iterator.rs

```diff
@@ -127,17 +127,23 @@ impl SstableStreamIterator {
     /// `self.block_iter` to `None`.
     async fn next_block(&mut self) -> HummockResult<()> {
         // Check if we want and if we can load the next block.
-        if self.remaining_blocks > 0 && let Some(block) = self.download_next_block().await? {
+        while self.remaining_blocks > 0 && let Some(block) = self.download_next_block().await? {
+            self.remaining_blocks -= 1;
+            let block_table_id = block.table_id();
+            if !self.existing_table_ids.contains(&block_table_id.table_id) {
+                continue;
+            }
```

Comment:
This optimization is already done in #8489. See `prune_from_valid_block_iter` for details.

Reply:
No...It seems we did not remove table-id from `SstableInfo` when unregister state tables